### ISSUE TITLE
[Bug]: Standalone panic with error `segment not found[segment=446469266327409774] [recovered]`

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#29398]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20231124-8a4cfb7d
- Deployment mode(standalone or cluster):standalone
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

[2023/12/21 08:07:23.855 +00:00] [ERROR] [conc/options.go:54] ["Conc pool panicked"] [panic="segment not found[segment=446469266327409774]"] [stack="github.com/milvus-io/milvus/pkg/util/conc.(*poolOption).antsOptions.func1\n\t/go/src/github.com/milvus-io/milvus/pkg/util/conc/options.go:54\ngithub.com/panjf2000/ants/v2.(*goWorker).run.func1.1\n\t/go/pkg/mod/github.com/panjf2000/ants/v2@v2.7.2/worker.go:54\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:884\ngithub.com/milvus-io/milvus/pkg/util/conc.(*Pool[...]).Submit.func1.1\n\t/go/src/github.com/milvus-io/milvus/pkg/util/conc/pool.go:72\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:884\ngithub.com/milvus-io/milvus/internal/datanode/writebuffer.(*writeBufferBase).getSyncTask.func2\n\t/go/src/github.com/milvus-io/milvus/internal/datanode/writebuffer/write_buffer.go:394\ngithub.com/milvus-io/milvus/internal/datanode/syncmgr.(*SyncTask).handleError\n\t/go/src/github.com/milvus-io/milvus/internal/datanode/syncmgr/task.go:77\ngithub.com/milvus-io/milvus/internal/datanode/syncmgr.(*SyncTask).Run\n\t/go/src/github.com/milvus-io/milvus/internal/datanode/syncmgr/task.go:129\ngithub.com/milvus-io/milvus/internal/datanode/syncmgr.(*keyLockDispatcher[...]).Submit.func1\n\t/go/src/github.com/milvus-io/milvus/internal/datanode/syncmgr/key_lock_dispatcher.go:34\ngithub.com/milvus-io/milvus/pkg/util/conc.(*Pool[...]).Submit.func1\n\t/go/src/github.com/milvus-io/milvus/pkg/util/conc/pool.go:79\ngithub.com/panjf2000/ants/v2.(*goWorker).run.func1\n\t/go/pkg/mod/github.com/panjf2000/ants/v2@v2.7.2/worker.go:67"]
panic: segment not found[segment=446469266327409774] [recovered]
	panic: segment not found[segment=446469266327409774] [recovered]
	panic: segment not found[segment=446469266327409774]
goroutine 251881 [running]:
panic({0x4c8f580, 0xc006062060})
	/usr/local/go/src/runtime/panic.go:987 +0x3bb fp=0xc003cd7788 sp=0xc003cd76c8 pc=0x19ebc5b
github.com/milvus-io/milvus/pkg/util/conc.(*poolOption).antsOptions.func1({0x4c8f580, 0xc006062060})
	/go/src/github.com/milvus-io/milvus/pkg/util/conc/options.go:56 +0x15b fp=0xc003cd7850 sp=0xc003cd7788 pc=0x336613b
github.com/panjf2000/ants/v2.(*goWorker).run.func1.1()
	/go/pkg/mod/github.com/panjf2000/ants/v2@v2.7.2/worker.go:54 +0x75 fp=0xc003cd78c8 sp=0xc003cd7850 pc=0x3363235
runtime.deferCallSave(0xc003cd7998, 0xc003cd7fb8?)
	/usr/local/go/src/runtime/panic.go:796 +0x88 fp=0xc003cd78d8 sp=0xc003cd78c8 pc=0x19eb848
runtime.runOpenDeferFrame(0xc00401e5a0)
	/usr/local/go/src/runtime/panic.go:769 +0x1b4 fp=0xc003cd7920 sp=0xc003cd78d8 pc=0x19eb674
panic({0x4c8f580, 0xc006062060})
	/usr/local/go/src/runtime/panic.go:884 +0x213 fp=0xc003cd79e0 sp=0xc003cd7920 pc=0x19ebab3
github.com/milvus-io/milvus/pkg/util/conc.(*Pool[...]).Submit.func1.1()
	/go/src/github.com/milvus-io/milvus/pkg/util/conc/pool.go:72 +0x97 fp=0xc003cd7a40 sp=0xc003cd79e0 pc=0x3ca71f7
runtime.deferCallSave(0xc003cd7b10, 0xc003cd7f50?)
	/usr/local/go/src/runtime/panic.go:796 +0x88 fp=0xc003cd7a50 sp=0xc003cd7a40 pc=0x19eb848
runtime.runOpenDeferFrame(0xc002f06c30)
	/usr/local/go/src/runtime/panic.go:769 +0x1b4 fp=0xc003cd7a98 sp=0xc003cd7a50 pc=0x19eb674
panic({0x4c8f580, 0xc006062060})
	/usr/local/go/src/runtime/panic.go:884 +0x213 fp=0xc003cd7b58 sp=0xc003cd7a98 pc=0x19ebab3
github.com/milvus-io/milvus/internal/datanode/writebuffer.(*writeBufferBase).getSyncTask.func2({0x545a900?, 0xc006062060?})
	/go/src/github.com/milvus-io/milvus/internal/datanode/writebuffer/write_buffer.go:394 +0x2e fp=0xc003cd7b78 sp=0xc003cd7b58 pc=0x3cb36ce
github.com/milvus-io/milvus/internal/datanode/syncmgr.(*SyncTask).handleError(...)
	/go/src/github.com/milvus-io/milvus/internal/datanode/syncmgr/task.go:77
github.com/milvus-io/milvus/internal/datanode/syncmgr.(*SyncTask).Run(0xc003e8c000)
	/go/src/github.com/milvus-io/milvus/internal/datanode/syncmgr/task.go:129 +0x7e2 fp=0xc003cd7e90 sp=0xc003cd7b78 pc=0x3ca0d22
github.com/milvus-io/milvus/internal/datanode/syncmgr.(*keyLockDispatcher[...]).Submit.func1()
	/go/src/github.com/milvus-io/milvus/internal/datanode/syncmgr/key_lock_dispatcher.go:34 +0xb1 fp=0xc003cd7f28 sp=0xc003cd7e90 pc=0x3ca7331
github.com/milvus-io/milvus/pkg/util/conc.(*Pool[...]).Submit.func1()
	/go/src/github.com/milvus-io/milvus/pkg/util/conc/pool.go:79 +0xbc fp=0xc003cd7f88 sp=0xc003cd7f28 pc=0x3ca70bc
github.com/panjf2000/ants/v2.(*goWorker).run.func1()
	/go/pkg/mod/github.com/panjf2000/ants/v2@v2.7.2/worker.go:67 +0x97 fp=0xc003cd7fe0 sp=0xc003cd7f88 pc=0x3363137
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1598 +0x1 fp=0xc003cd7fe8 sp=0xc003cd7fe0 pc=0x1a259e1
created by github.com/panjf2000/ants/v2.(*goWorker).run
	/go/pkg/mod/github.com/panjf2000/ants/v2@v2.7.2/worker.go:48 +0x65

### Steps To Reproduce

```markdown
hard to tell which step caused this issue.
because I send many requests to Milvus at the same time.
```

### Milvus Log

[panic.log](https://github.com/milvus-io/milvus/files/13738906/panic.log)
full logs can go to loki

cluster: 4am
namespace: chaos-testing
release name: cdc-test-upstream-80

**Comment by xiaofan-luan** [comment]
what milvus version it is?
is it 2.3 or 2.3 latest?

**Comment by zhuwenxing** [comment]
@xiaofan-luan 
made a mistake, it was a master image tag `master-20231124-8a4cfb7d`

**Comment by smellthemoon** [comment]
panic because  write buffer has use compacted segment as candidate. fix with #28816
plz help to verify. @zhuwenxing

**Comment by yanliang567** [comment]
/assign @zhuwenxing 
please help to verify the fix
